src/graph/graph.c

```diff
@@ -379,9 +379,9 @@ void Graph_DeleteEdge(Graph *g, int src_id, int dest_id) {
 void Graph_LabelNodes(Graph *g, int start_node_id, int end_node_id, int label, NodeIterator **it) {
     assert(g &&
            start_node_id < g->node_count &&
-           start_node_id >=0 &&
-           start_node_id < end_node_id &&
-           end_node_id < g->node_count);
+           start_node_id >= 0 &&
+           start_node_id <= end_node_id &&
+           end_node_id <= g->node_count);
```

Comment:
I think end_node_id must be smaller then g->node_count.
if end_node_id equals g->node_count then we would try to access an invalid index within the loop below.